src/plugins/discover/public/application/main/components/top_nav/open_alerts_popover.tsx

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useCallback, useState, useMemo } from 'react';
+import ReactDOM from 'react-dom';
+import { I18nStart } from 'kibana/public';
+import { EuiWrappingPopover, EuiLink, EuiContextMenu, EuiToolTip } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { ISearchSource } from '../../../../../../data/common';
+import { KibanaContextProvider } from '../../../../../../kibana_react/public';
+import { DiscoverServices } from '../../../../build_services';
+import { updateSearchSource } from '../../utils/update_search_source';
+import { useDiscoverServices } from '../../../../utils/use_discover_services';
+
+const container = document.createElement('div');
+let isOpen = false;
+
+const ALERT_TYPE_ID = '.es-query';
+
+interface AlertsPopoverProps {
+  onClose: () => void;
+  anchorElement: HTMLElement;
+  searchSource: ISearchSource;
+}
+
+export function AlertsPopover(props: AlertsPopoverProps) {
+  const dataView = props.searchSource.getField('index')!;
+  const searchSource = props.searchSource;
+  const services = useDiscoverServices();
+  const { triggersActionsUi } = services;
+  const [alertFlyoutVisible, setAlertFlyoutVisibility] = useState(false);
+
+  const onCloseAlertFlyout = useCallback(() => props.onClose(), [props]);
+
+  /**
+   * Provides the default parameters used to initialize the new rule
+   */
+  const getParams = useCallback(() => {
+    const nextSearchSource = searchSource.createCopy();
+    updateSearchSource(nextSearchSource, true, {
+      indexPattern: searchSource.getField('index')!,
+      services,
+      sort: [],
+      useNewFieldsApi: true,
```

Comment:
is this intended to always work exclusively with the fields api?

Reply:
tbh, if you look at `updateSearchSource` function, `useNewFieldsApi` parameter will affect nothing, because the second one (`persist`) is set to true.